ha/core/system_health/bootstrap/json_kv_convertor.py

```diff
@@ -43,85 +52,170 @@ def __init__(self):
+                 },
+                 {
+                     storage: []
+                 }
+        }
+        This routine further handles the inside hierarchy of compute and storage.
+        Args:
+            key(str): node health key
+            component(list or dict): node health component
+        """
+        if isinstance(component, list) and component:
+            for res_comp in component:
+                if 'uid' in res_comp and 'last_updated' in res_comp:
+                    val = res_comp['uid']
+                    self._update_health(f'{key}.uid', val)
+                    val = res_comp['last_updated']
+                    self._update_health(f'{key}.last_updated', val)
+                if 'health' in res_comp:
+                    new_key = key + '.health.status'
+                    val = res_comp['health']['status']
+                    self._update_health(new_key, val)
+        else:
+            if isinstance(component, dict) and component:
+                for sub_comp in component:
+                    new_key = key + f'.{sub_comp.lower()}'
+                    self._parse_health_comp_dict(component[sub_comp], new_key)
 
     # parse json and generate health events
-    def generate_health(self, json_file_nm, conf_index, conf_store):
-
+    def generate_health(self, json_file_nm: str, conf_index, conf_store) -> None:
+        """
+        Load the json strctured node health data received from Discovery module
+        to confstore and then retrives the required data using index and key
+        Args:
+            json_file_nm(str): node health json file to be parsed
+            conf_index(str): index for common conf store
+            conf_store(obj): conf store object
+        """
         self._conf_index = conf_index
         self._conf_store = conf_store
+        node_health_list = []
 
         try:
-            with open(json_file_nm, "r") as fi:
-                schema = json.load(fi)
+            Conf.load(NODE_HEALTH_CONF_INDEX, f'json://{json_file_nm}')
+            node_health_list = Conf.get(NODE_HEALTH_CONF_INDEX, NODE_DATA_KEY)
+            # Specifically to retrieve upper_level keys such as:
+            # node.compute.<'uid', 'last_updated', 'health.status'>
+            # node.storage.<'uid', 'last_updated', 'health.status'>
+            for node_comp in node_health_list:
+                if node_health_list[node_comp]:
+                    uid = node_health_list[node_comp][0]['uid']
+                    self._update_health(f'node.{node_comp}.uid', uid)
+                    last_updated = node_health_list[node_comp][0]['last_updated']
+                    self._update_health(f'node.{node_comp}.last_updated', last_updated)
+                    health_status = node_health_list[node_comp][0]['health']['status']
+                    self._update_health(f'node.{node_comp}.health.status', health_status)
+
+            # Get the node.compute[0] and node.storage[0] data and iterate over it
```

Comment:
The function is currently expecting a fixed level of hierarchy for the data points (last_modified, uid, status). 
Tomorrow if the level of hierarchy changes for any component (ex: maybe due to different hardware being used)
it will not be able to update health for all the components.

Please check if you can use the old code from _walk function to do this, The _walk function can handle any level of hierarchy for any component

If its too complicated, please add a TODO here mentioning the same and it can be taken up later.

@indrajitzagade  any inputs from you ?

Reply:
Added a TODO for now.

Reply:
TODO is good for now.